### PR TITLE
chore: improve tooling for reordering vale substitution rules

### PR DESCRIPTION
**Pull request by ccoVeille**

### CODE REVIEW

**Comment by ccoVeille** [comment]
@alexvonme @tbazant maybe you would have interest for this in your other projects

I had to play a lot with `yq`, but I managed to make it works

**Comment by ccoVeille** [comment]
@alexvonme feel free to merge this one once you will have read it and are fine and with it

**Comment by alexvonme** [comment]
I think this one should be closed since we're now migrating these types of descriptions to ``/sources``.

**Comment by ccoVeille** [comment]
Not closed, but rewritten and pushed back, as the tips is now in another file

**Comment by ccoVeille** [comment]
Please review

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -101,3 +101,68 @@ This simply ensures that you have the legal right to make the contribution.
...
+```shell
+markdownlint --fix .
+```
+
+### ls-lint
+
+Validating the file name
+
+- install [markdownlint](https://github.com/loeffel-io/ls-lint)
+- launch `ls-lint`
```
Comment (alexvonme) [inline CONTRIBUTING.md:134]:
```suggestion
### ls-lint

Validating the file name

- install [ls-lint](https://github.com/loeffel-io/ls-lint)
- launch `ls-lint`
```

**Comment by alexvonme** [comment]
@ccoVeille I think it would be better if we had ``Tooling`` right before ``Credit``. It's generally accepted that the credit section come last in a guide.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -101,3 +101,68 @@ This simply ensures that you have the legal right to make the contribution.
 
 Please feel free to add yourself to the CONTRIBUTORS.md file with your pull request.
 We appreciate your efforts and would love to add you to our list of contributors.
+
+## Tooling
+
+### yamllint
```
Comment (alexvonme) [inline CONTRIBUTING.md:107]:
```suggestion
## Tooling

We use linters to standardize and maintain style in our files. 
These linters are added as GitHub actions to verify that each pull request is compatible.
In this section, we provide instructions to detail how you can also run these linters locally.

In addition, we also describe the steps to deconstruct our rules to allow their adoption for other use cases.

### yamllint
```

Adding a description here might be useful.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -101,3 +101,74 @@ This simply ensures that you have the legal right to make the contribution.
 
 Please feel free to add yourself to the CONTRIBUTORS.md file with your pull request.
 We appreciate your efforts and would love to add you to our list of contributors.
+
+## Tooling
+
+We use linters to standardize and maintain style in our files. 
+These linters are added as GitHub actions to verify that each pull request is compatible.
```
Comment (ccoVeille) [inline CONTRIBUTING.md:108]:
```suggestion
These linters are added as GitHub Actions to verify that each pull request is compatible.
```

**Comment by ccoVeille** [comment]
> @ccoVeille I think it would be better if we had ``Tooling`` right before ``Credit``. It's generally accepted that the credit section come last in a guide.

I hesitated, yes.

Thanks for the suggested changes by the way

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -101,3 +101,74 @@ This simply ensures that you have the legal right to make the contribution.
...
+
+Validating and fixing the errors in your Markdown files
+
+- install [markdownlint](https://github.com/markdownlint/markdownlint)
+- launch `markdownlint`
+
+```shell
+markdownlint --fix .
+```
+
```
Comment (alexvonme) [inline CONTRIBUTING.md:134]:
```suggestion

You can also use the [VS Code extension](https://github.com/DavidAnson/vscode-markdownlint) based on the [markdownlint library](https://github.com/DavidAnson/markdownlint) for Node.js.

```

We could add this here to let people know about the VS Code extension

**Review by alexvonme — APPROVED** [review]
I'm okay with the merge after we add the VS Code extension link as well.

Thanks for explanations here.

File: CONTRIBUTING.md (hunk 4 above)
Comment (ccoVeille) [reply CONTRIBUTING.md:134]: Good point

**Comment by ccoVeille** [comment]
> I'm okay with the merge after we add the VS Code extension link as well.
> 
> Thanks for explanations here.

I also plan to use VS Code native way to suggest extension when opening the project

- https://github.com/jargonLint/jargonLint/issues/55

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -101,3 +101,75 @@ This simply ensures that you have the legal right to make the contribution.
...
+## Tooling
+
+We use linters to standardize and maintain style in our files. 
+These linters are added as GitHub Actions to verify that each pull request is compatible.
+In this section, we provide instructions to detail how you can run these linters locally.
+
+In addition, we also describe the steps to deconstruct the rules to help their adoption for other use cases.
+
+### yamllint
```
Comment (ccoVeille) [inline CONTRIBUTING.md:113]: I forgot typos

https://github.com/crate-ci/typos

And the related VScode extension

https://marketplace.visualstudio.com/items?itemName=tekumara.typos-vscode

Comment (alexvonme) [reply CONTRIBUTING.md]: I added that one.

**Pull request merged**